tealium/collectors/attribution/TealiumAdClient.swift

```diff
@@ -25,7 +36,54 @@ public class TealiumAdClient: TealiumAdClientProtocol {
     }
 
     public func requestAttributionDetails(_ completionHandler: @escaping ([String: NSObject]?, Error?) -> Void) {
```

Comment:
As said previously, probably it's worth changing this to handle the PersistentAppData creation itself.
Probably is also worth having an @available(iOS 14.3, *) method that does the HTTP call and maps and another that does it for the older devices and maps, otherwise this method will also be too big.

Reply:
I feel like this could still be improved a little since the method now has the implementation of the URLSession inside for one of the two ifs, making it harder to read and realise there is more stuff at the end that is about the other if.

Actually, looking better at this class, this conforms to a protocol: `TealiumAdClientProtocol`.
This protocol just offers one method, and a static getter.
This is the perfect case for splitting the Inner work in two classes:
One AdClient that uses the old API and just maps the return to the caller.
One new AdClient (HTTPRequest Ad Client?) that makes the URLSession API request.
Both have to return the attributionData and one doesn't need to instanciate the other dependencies (like ADClient.shared() for example or the URLSession if we decide no to use the shared one).

This would also be very useful for testing, as we will be able to pass the new and the old client separately and have no constraints on which version of the OS are we in (as long as it's over 14.3) and will work for both the classes if we have separate tests for both.